apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryHandler.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2022 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Storage.v1;
+using Google.Apis.Http;
+using Google.Apis.Json;
+using Google.Apis.Util;
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Storage.V1
+{
+    /// <summary>
+    /// Handler for exponential backoff for request messages which permit retries.
+    /// This is similar to BackoffHandler in Google.Apis.Http, but that doesn't give us enough flexibility.
+    /// This handler does not enforce a maximum number of retries; that's configured via <see cref="ConfigurableMessageHandler.NumTries"/>.
+    /// </summary>
+    internal sealed class RetryHandler : IHttpUnsuccessfulResponseHandler
+    {
+        private static readonly string[] s_retriableErrors = { "rateLimitExceeded", "backendError" };
```

Comment:
This is retrying based on the reason, which is what we were advised for BigQuery. Is that the same for Storage, or should we use the code instead?